feat(tooltip): Refactor to a simpler API (#528)
Workday/canvas-kit: modules/tooltip/react/README.md

```tsx
import * as React from 'react';

import {Tooltip} from '@workday/canvas-kit-react-tooltip';
import {IconButton} from '@workday/canvas-kit-react-button';

const TooltipExample = () => {
  return (
    <Tooltip title="Close">
      <IconButton variant={IconButton.Variant.Circle} icon={xIcon} aria-label="Close" />
    </Tooltip>
  );

> };

Author: Wow! Isn't this simpler?!

### Required

#### title: string | React.ReactNode

> > If present, this will switch the Tooltip into a simpler to use mode where the Tooltip component
> > can wrap a target and `title` is the contents of the tooltip. This should be a string in most

Reviewer: Isn't this only determined by which component you use? Omitting `title` will not give you the old behavior (and the type requires `title` anyway)

```suggestion
> The text contents of the tooltip. This should be a string in most
```

Author: Ah, yes. Remnants of wip

> cases. HTML is supported, but only text is understood by assistive technology. This is true for
> both `label` and `describe` modes.

#### children: React.ReactNode

> > The contents of the target for the Tooltip.

Reviewer: It's not really the contents. If `children` is a button and I say "the button's contents", I think of the text, not the outer element

```suggestion
> The target (anchor element) for the Tooltip.
```

Author: That is more clear

>
> **Note:** This **must** be an Element, StyledComponent or any other component that forwards extra
> props to an Element. Tooltip works running `React.cloneElement` on the children and adds extra
> properties like aria attributes and event handlers. This is currently a limitation of the Tooltip
> component. Functionality will not work if this condition isn't met

---

### Optional

#### type: 'label' | 'describe'

> > If present, this will switch the Tooltip into a simpler to use mode where the Tooltip component

Reviewer: Same here
